### ISSUE TITLE
Does RxJS has something more powerful than ordering for Angular?

### ISSUE THREAD

**Issue by shystruk** [OasisDigital/angularjs-rxjs-example#1]
Hi @kylecordes ,
If you have used RxJS with Angular, could you please share what is the main benefit?
Ordering you may handle by native Promise as well ;)
```javascript
function getUserInfo(userName) {
    let lastSelectedUser = userName

    http.get().then((user => {
		if (lastSelectedUser === user.name) {
			renderUser(user)
                }
	})
}
```

**Comment by kylecordes** [comment]
@shystruk I talk about this more in the video that goes with this repo, https://www.youtube.com/watch?v=XY9VPrKuJaA

For your example code snippet above - yes, for simple cases, you can easily write ad-hoc code to handle this specific problem with. But this manual coding solution does not scale well at all, to more complex applications where the code to load data may be farther away from the UI. I have found RxJS is much better than the ad hoc approach, both for this simple case and for more complex data flows.

**Comment by shystruk** [comment]
Thanks for the response Kyle

On Dec 15, 2017 16:41, "Kyle Cordes" <notifications@github.com> wrote:

Closed #1 <https://github.com/OasisDigital/angularjs-rxjs-example/issues/1>.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/OasisDigital/angularjs-rxjs-example/issues/1#event-1389725413>,
or mute the thread
